documentation/create_overview_doc/createdoc.sh

```diff
@@ -111,419 +82,215 @@ function read_skip_list() {
-         local ret_code=$?
-
-        if ! bito_response_ok "$ret_code" "$bito_output"; then
-            echo "Attempt $attempt: bito command for module: $current_module failed or did not return enough content. Retrying in $RETRY_DELAY seconds..." >&2
-            sleep $RETRY_DELAY
-            ((attempt++))
-        else
-            echo "Attempt $attempt: bito command for module: $current_module succeeded with sufficient content." >&2
-            echo "$bito_output"
-            update_token_usage "$combined_output" "$bito_output"
-            return 0
-        fi
-    done
-
-    echo "Failed to call bito for module: $current_module after $MAX_RETRIES attempts with adequate content." >&2
-    return 1
-}
-
-function fix_mermaid_syntax() {
-    local mermaid_content="$1"
-    local fixed_mermaid_content
-
-    # Combine removal of empty parentheses and quotations in one step
-    fixed_mermaid_content=$(echo "$mermaid_content" | sed 's/()//g; s/"//g')
-
-    # Insert space between an opening square bracket '[' and a forward slash '/'
-    fixed_mermaid_content=$(echo "$fixed_mermaid_content" | sed 's/\[\//[ \//g')
-
-    # Adjust nested square brackets '[]'
-    fixed_mermaid_content=$(echo "$fixed_mermaid_content" | sed 's/\[\([^]]*\)\[\([^]]*\)\]\]/[\1 \2]/g')
-
-    echo "$fixed_mermaid_content"
-}
-
-# Function to fix Mermaid diagram syntax using bito AI
-function fix_mermaid_syntax_with_bito() {
-    local fixed_mermaid_content
-    fixed_mermaid_content=$(echo "$mermaid_content" | bito -p "$prompt_folder/mermaid_doc_prompt.txt" | awk '/^```mermaid$/,/^```$/{if (!/^```mermaid$/ && !/^```$/) print}')
-    
-    local ret_code=$?
-    if ! bito_response_ok "$ret_code" "$fixed_mermaid_content"; then
-        echo "Error in bito response for fixing mermaid syntax with bito."
+    if bito_output=$(call_bito_with_retry "$combined_output" "$prompt_file_path" "Failed for module: $current_module"); then
+        echo "Bito call successful for module: $current_module" >&2
+        echo "$bito_output"
+        update_token_usage "$combined_output" "$bito_output"
+    else
+        echo "Failed to call bito for module: $current_module with adequate content." >&2
         return 1
     fi
 }
 
-# Function to validate Mermaid diagram syntax
-function validate_mermaid_syntax() {
-    local mermaid_content="$1"
-    local temp_mmd_file=$(mktemp)
-    # Write Mermaid content to a temporary file
-    echo "$mermaid_content" > "$temp_mmd_file"
-    # Attempt to parse the Mermaid diagram using mmdc
-    local output=$(mmdc -i "$temp_mmd_file" -o /dev/null 2>&1)
-    local status=$?
-    # Clean up the temporary file
-    rm "$temp_mmd_file"
-    if [ $status -ne 0 ]; then
-        echo "Mermaid syntax validation failed. Please check the diagram syntax." >&2
-        echo "$output" >&2
-        return 1
-    fi
-    return 0
-}
+function create_find_command() {
+    local lang_file="$1"
+    local folder_to_document="$2"
```

Comment:
**Optimization Issue**: The script's approach to handling retry logic and error messages is verbose and lacks efficiency. The retry mechanism is manually coded in multiple places, which increases the risk of errors and code duplication. <br> **Fix**: Implement a generic retry function that accepts commands to execute, a maximum retry count, and a delay between retries. This function should be used throughout the script to handle retries in a DRY (Don't Repeat Yourself) manner. <br> **Code Suggestion**: 
 ```
 function generic_retry() {
     local command_to_execute="$1"
     local max_retries=$2
     local delay=$3
     local attempt=1

     until [ $attempt -gt $max_retries ]; do
         eval $command_to_execute
         retval=$?
         if [ $retval -eq 0 ]; then
             return 0
         fi
         echo "Attempt $attempt failed... retrying in $delay seconds"
         sleep $delay
         ((attempt++))
     done
     echo "Failed after $max_retries attempts."
     return 1
 }
 ```